xarray/backends/pydap_.py

```diff
@@ -49,36 +51,37 @@ def __getitem__(self, key):
         )
 
     def _getitem(self, key):
-        # pull the data from the array attribute if possible, to avoid
-        # downloading coordinate data twice
-        array = getattr(self.array, "array", self.array)
```

Comment:
don't we need this any more?

Reply:
I set the `output_grid=False` by default on pydap, so that [should not be necessary](https://pydap.github.io/pydap/en/DAP_Protocol.html#old-dap2-approach). But I'll double check (All the tests passed on the xarray-tests environment)

```
pytest -v xarray/tests/test_backends.py --run-network-tests
```

Reply:
I updated/upgraded to a newer pydap version on `min_all_dep.yml`. The failing test `test_cmp_local_file` is gone as expected. Meaning this is correct. I am also removing `output_grid` as an argument. It is not really used. So I added a deprecation warning.